fdbserver/KeyValueStoreShardedRocksDB.actor.cpp

```diff
@@ -0,0 +1,1839 @@
+	case rocksdb_env_lock_file_nanos:
+		return rocksdb::get_perf_context()->env_lock_file_nanos;
+	case rocksdb_env_unlock_file_nanos:
+		return rocksdb::get_perf_context()->env_unlock_file_nanos;
+	case rocksdb_env_new_logger_nanos:
+		return rocksdb::get_perf_context()->env_new_logger_nanos;
+	default:
+		break;
+	}
+	return 0;
+}
+
+ACTOR Future<Void> rocksDBAggregatedMetricsLogger(std::shared_ptr<RocksDBMetrics> rocksDBMetrics, rocksdb::DB* db) {
+	loop {
+		wait(delay(SERVER_KNOBS->ROCKSDB_METRICS_DELAY));
+		/*
+		if (SERVER_KNOBS->ROCKSDB_ENABLE_STATISTIC) {
+		    rocksDBMetrics->logStats(db);
+		}
+		if (SERVER_KNOBS->ROCKSDB_PERFCONTEXT_SAMPLE_RATE != 0) {
+		    rocksDBMetrics->logPerfContext(true);
+		}*/
+	}
+}
+
+struct ShardedRocksDBKeyValueStore : IKeyValueStore {
+	using CF = rocksdb::ColumnFamilyHandle*;
+	using CommitBatchMap = std::unordered_map<std::shared_ptr<DataShard>, std::unique_ptr<rocksdb::WriteBatch>>;
+
+	struct Writer : IThreadPoolReceiver {
+		int threadIndex;
+		std::shared_ptr<RocksDBMetrics> rocksDBMetrics;
+		std::shared_ptr<rocksdb::RateLimiter> rateLimiter;
+
+		explicit Writer(int threadIndex, std::shared_ptr<RocksDBMetrics> rocksDBMetrics)
+		  : threadIndex(threadIndex), rocksDBMetrics(rocksDBMetrics),
+		    rateLimiter(SERVER_KNOBS->ROCKSDB_WRITE_RATE_LIMITER_BYTES_PER_SEC > 0
+		                    ? rocksdb::NewGenericRateLimiter(
+		                          SERVER_KNOBS->ROCKSDB_WRITE_RATE_LIMITER_BYTES_PER_SEC, // rate_bytes_per_sec
+		                          100 * 1000, // refill_period_us
+		                          10, // fairness
+		                          rocksdb::RateLimiter::Mode::kWritesOnly,
+		                          SERVER_KNOBS->ROCKSDB_WRITE_RATE_LIMITER_AUTO_TUNE)
+		                    : nullptr) {}
+
+		~Writer() override {}
+
+		void init() override {}
+
+		struct OpenAction : TypedAction<Writer, OpenAction> {
+			ShardManager* shardManager;
+			ThreadReturnPromise<Void> done;
+			Optional<Future<Void>>& metrics;
+			const FlowLock* readLock;
+			const FlowLock* fetchLock;
+			std::shared_ptr<RocksDBErrorListener> errorListener;
+
+			OpenAction(ShardManager* shardManager,
+			           Optional<Future<Void>>& metrics,
+			           const FlowLock* readLock,
+			           const FlowLock* fetchLock,
+			           std::shared_ptr<RocksDBErrorListener> errorListener)
+			  : shardManager(shardManager), metrics(metrics), readLock(readLock), fetchLock(fetchLock),
+			    errorListener(errorListener) {}
+
+			double getTimeEstimate() const override { return SERVER_KNOBS->COMMIT_TIME_ESTIMATE; }
+		};
+
+		void action(OpenAction& a) {
+			auto status = a.shardManager->init();
+			if (!status.ok()) {
+				logRocksDBError(status, "Open");
+				a.done.sendError(statusToError(status));
+				return;
+			}
+
+			TraceEvent(SevInfo, "RocksDB").detail("Method", "Open");
+			a.done.send(Void());
+		}
+
+		struct DeleteVisitor : public rocksdb::WriteBatch::Handler {
```

Comment:
Can you update this to: https://github.com/apple/foundationdb/blob/f00606fd71c884a066d71eb5632e3f81d460c882/fdbserver/KeyValueStoreRocksDB.actor.cpp#L1003-L1036?

Reply:
I removed DV in https://github.com/apple/foundationdb/pull/7184. I'll add it back when adding range clear.